src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/google/GoogleExceptionSupport.scala

```diff
@@ -0,0 +1,37 @@
+package org.broadinstitute.dsde.workbench.leonardo.dao.google
+
+import akka.http.scaladsl.model.StatusCodes
+import com.google.api.client.googleapis.json.GoogleJsonResponseException
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.leonardo.model.{Cluster, LeoException}
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * Adds support for recovering from Google API errors and re-throwing LeoExceptions.
+  */
+object GoogleExceptionSupport extends LazyLogging {
```

Comment:
I didn't want `HttpGoogleDataprocDAO` to throw `LeoExceptions` so I added this extension class to catch `GoogleJsonResponseException`s and convert to `CallToGoogleApiFailedException`. This allows us to do this in various places in the code:
```
dataprocDAO.createCluster(...).handleGoogleException(..)
```
I'm still deciding how annoying that is...

Reply:
Hm, yeah it might be better to make the DAO throw `WorkbenchException`, and deal with the `WorkbenchException`s in Leo code (either in the routes or downstream).